pkg/local_object_storage/shard/control.go

```diff
@@ -161,82 +161,93 @@ func (s *Shard) resyncMetabase() error {
 		return fmt.Errorf("could not reset metabase: %w", err)
 	}
 
-	obj := objectSDK.New()
-
-	err = blobstor.IterateBinaryObjects(s.blobStor, func(addr oid.Address, data []byte, descriptor []byte) error {
-		if err := obj.Unmarshal(data); err != nil {
-			s.log.Warn("could not unmarshal object",
-				zap.Stringer("address", addr),
-				zap.String("err", err.Error()))
-			return nil
+	if s.writeCache != nil {
+		// ensure there will not be any raсes in write-cache -> blobstor object
+		// backgroung flushing while iterating blobstor
+		err = s.writeCache.Flush(true)
```

Comment:
We're asking for a metabase resync and getting a cache flush. It's somewhat unobvious. Can we have caches/stores RO and perform resync as is?